bug: invalid memory address or nil pointer dereference

Run terrafom plan

```
 Error: Request cancelled
│ 
│ The plugin6.(*GRPCProvider).ReadDataSource request was cancelled.
╵

Stack trace from the terraform-provider-sentry_v0.14.6 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x100e97b8c]

goroutine 11 [running]:
github.com/jianyuan/terraform-provider-sentry/internal/provider.(*ProjectDataSource).Read(0x1400014cb20, {0x1014088e8, 0x14000394d80}, {{{{0x10140eef8, 0x140001961e0}, {0x1012a43e0, 0x14000196090}}, {0x101410cd0, 0x14000390f80}}, {{{0x0, ...}, ...}, ...}, ...}, ...)
        github.com/jianyuan/terraform-provider-sentry/internal/provider/data_source_project.go:138 +0x31c
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ReadDataSource(0x1400015f688, {0x1014088e8, 0x14000394d80}, 0x140000b8a80, 0x1400069d550)
        github.com/hashicorp/terraform-plugin-framework@v1.15.0/internal/fwserver/server_readdatasource.go:103 +0x638
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ReadDataSource(0x1400015f688, {0x1014088e8?, 0x14000394c90?}, 0x14000394960)
        github.com/hashicorp/terraform-plugin-framework@v1.15.0/internal/proto6server/server_readdatasource.go:55 +0x2e0
github.com/hashicorp/terraform-plugin-mux/tf6muxserver.(*muxServer).ReadDataSource(0x140002d2180, {0x1014088e8?, 0x140003949c0?}, 0x14000394960)
        github.com/hashicorp/terraform-plugin-mux@v0.20.0/tf6muxserver/mux_server_ReadDataSource.go:36 +0x184
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ReadDataSource(0x14000296460, {0x1014088e8?, 0x14000119c20?}, 0x1400042ae60)
        github.com/hashicorp/terraform-plugin-go@v0.28.0/tfprotov6/tf6server/server.go:733 +0x1cc
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ReadDataSource_Handler({0x1013d72c0, 0x14000296460}, {0x1014088e8, 0x14000119c20}, 0x140002d2c80, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.28.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:741 +0x1c0
google.golang.org/grpc.(*Server).processUnaryRPC(0x14000288200, {0x1014088e8, 0x14000119b90}, 0x14000400300, 0x14000394180, 0x101abef70, 0x0)
        google.golang.org/grpc@v1.72.1/server.go:1405 +0xca8
google.golang.org/grpc.(*Server).handleStream(0x14000288200, {0x1014094e0, 0x14000476000}, 0x14000400300)
        google.golang.org/grpc@v1.72.1/server.go:1815 +0x910
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.72.1/server.go:1035 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 39
        google.golang.org/grpc@v1.72.1/server.go:1046 +0x13c

Error: The terraform-provider-sentry_v0.14.6 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

Versions
```
terraform version
Terraform v1.12.1
on darwin_arm64
+ provider registry.terraform.io/hashicorp/aws v6.21.0
+ provider registry.terraform.io/hashicorp/local v2.6.1
+ provider registry.terraform.io/hashicorp/null v3.2.4
+ provider registry.terraform.io/hashicorp/random v3.7.2
+ provider registry.terraform.io/jianyuan/sentry v0.14.6
+ provider registry.terraform.io/kreuzwerker/docker v3.6.2
+ provider registry.terraform.io/metio/git v2025.11.14
+ provider registry.terraform.io/petoju/mysql v3.0.84
+ provider registry.terraform.io/rutledgepaulv/counters v0.0.5

Your version of Terraform is out of date! The latest version
is 1.13.5. You can update by downloading from https://developer.hashicorp.com/terraform/install
```

--- a/internal/provider/data_source_client_key.go
+++ b/internal/provider/data_source_client_key.go
@@ -144,7 +144,7 @@ func (d *ClientKeyDataSource) Read(ctx context.Context, req datasource.ReadReque
 				resp.Diagnostics.Append(diagutils.NewClientError("read", err))
 				return
 			}
-			if httpResp.StatusCode() != http.StatusOK {
+			if httpResp.StatusCode() != http.StatusOK || httpResp.JSON200 == nil {
 				resp.Diagnostics.Append(diagutils.NewClientStatusError("read", httpResp.StatusCode(), httpResp.Body))
 				return
 			}

--- a/internal/provider/data_source_project.go
+++ b/internal/provider/data_source_project.go
@@ -130,7 +130,7 @@ func (d *ProjectDataSource) Read(ctx context.Context, req datasource.ReadRequest
 		resp.Diagnostics.Append(diagutils.NewNotFoundError("project"))
 		resp.State.RemoveResource(ctx)
 		return
-	} else if httpResp.StatusCode() != http.StatusOK {
+	} else if httpResp.StatusCode() != http.StatusOK || httpResp.JSON200 == nil {
 		resp.Diagnostics.Append(diagutils.NewClientStatusError("read", httpResp.StatusCode(), httpResp.Body))
 		return
 	}

--- a/internal/provider/resource_project.go
+++ b/internal/provider/resource_project.go
@@ -878,7 +878,7 @@ func (r *ProjectResource) removeDefaultKey(ctx context.Context, organization str
 		)
 		if err != nil {
 			return err
-		} else if httpResp.StatusCode() != http.StatusOK {
+		} else if httpResp.StatusCode() != http.StatusOK || httpResp.JSON200 == nil {
 			return fmt.Errorf("unable to list project client keys, got status %d: %s", httpResp.StatusCode(), string(httpResp.Body))
 		}
 

